tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/PortalRuntime.java

```diff
@@ -21,22 +21,28 @@
 import java.beans.PropertyChangeListener;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
 
 import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
 import org.eclipse.jdt.launching.IRuntimeClasspathEntry;
 import org.eclipse.jdt.launching.IVMInstall;
+import org.eclipse.jdt.launching.IVMInstall2;
 import org.eclipse.jdt.launching.JavaRuntime;
+import org.eclipse.osgi.util.NLS;
 import org.eclipse.wst.server.core.model.RuntimeDelegate;
 
 
 /**
  * @author Gregory Amerson
+ * @author Simon Jiang
  */
 public class PortalRuntime extends RuntimeDelegate implements ILiferayRuntime, PropertyChangeListener
 {
+    private static Map<String, Integer> javaVersionMap = new ConcurrentHashMap<String, Integer>();
```

Comment:
No need for this map, the version parsing check is fast enough calling this code multiple times should not be a problem.